### PR TITLE
Added the feature to find the normalized loan period

### PR DESCRIPTION
**Pull request by chakshuahuja**
to find the bottleneck of the availability of books.

Also, fixed some indentation in modules/webstyle/lib/webinterface_layout.py and removed conflict markers in this file.

### CODE REVIEW

File: modules/bibcirculation/lib/bibcirculation_dblayer.py [hunk 1]
```diff
@@ -591,7 +591,46 @@ def get_pdf_request_data(status):
 ###
 
 
+def calculate(s, e, n):
```
Comment (egabancho) [inline modules/bibcirculation/lib/bibcirculation_dblayer.py:594]: I think you shouldn't use one character variables, characters are for free :stuck_out_tongue_winking_eye:

File: modules/bibcirculation/lib/bibcirculation_dblayer.py [hunk 2]
```diff
@@ -591,7 +591,46 @@ def get_pdf_request_data(status):
...
+    @type e: string
+
+    @param n: number of top results.
+    @type n: string
+
+    @return list with Record No, Loan Period,
+    Item copies, Volume and Item List.
+    """
+    s = str(s)
+    e = str(e)
```
Comment (egabancho) [inline modules/bibcirculation/lib/bibcirculation_dblayer.py:612]: do you need this two here? If you say they are strings in the doc string, ti should be enough I guess

File: modules/bibcirculation/lib/bibcirculation_dblayer.py [hunk 3]
```diff
@@ -591,7 +591,46 @@ def get_pdf_request_data(status):
...
+    from crcLOANREQUEST, crcITEM
+    where period_of_interest_from > %s
+    and
+    period_of_interest_to < %s
+    and
+    crcITEM.barcode = crcLOANREQUEST.barcode
+    group by crcLOANREQUEST.id_bibrec
+    order by NormalizedLoanPeriod desc
+    limit %s
+    """, [s, e, n])
```
Comment (egabancho) [inline modules/bibcirculation/lib/bibcirculation_dblayer.py:631]: It is better if this is a tuple  `(s, e, n)`

File: modules/bibcirculation/lib/bibcirculation_dblayer.py [hunk 4]
```diff
@@ -591,7 +591,46 @@ def get_pdf_request_data(status):
...
+    @param n: number of top results.
+    @type n: string
+
+    @return list with Record No, Loan Period,
+    Item copies, Volume and Item List.
+    """
+    s = str(s)
+    e = str(e)
+    n = int(n)
+    res = run_sql("""select crcLOANREQUEST.id_bibrec as Record_no,
```
Comment (egabancho) [inline modules/bibcirculation/lib/bibcirculation_dblayer.py:614]: Can you maybe format a bit the query so it is easier to read?

File: modules/bibcirculation/lib/bibcirculation_templates.py [hunk 5]
```diff
@@ -1970,6 +1971,149 @@ def tmpl_borrower_search_result(self, result, redirect_to_new_request=False,
...
+              <tr>
+                <th>%s</th>
+                <th>%s</th>
+                <th>%s</th>
+                <th>%s</th>
+		<th>%s</th>
+              </tr>
+            </thead>
+          <tbody>
+        """ % (len(result), _("Recorrd No"), _("No of items"),
```
Comment (egabancho) [inline modules/bibcirculation/lib/bibcirculation_templates.py:2077]: Can you fix a bit the indentation?

File: modules/bibcirculation/lib/bibcirculation_templates.py [hunk 6]
```diff
@@ -1970,6 +1971,149 @@ def tmpl_borrower_search_result(self, result, redirect_to_new_request=False,
...
+	 _("Loan Period(in days)"), _("Item List"), 
+               _("Volume"))
+
+            for recid in result:
+
+                (volume, loanPeriod, itemList, noOfItems, recordNo) = recid
```
Comment (egabancho) [inline modules/bibcirculation/lib/bibcirculation_templates.py:2083]: I think you can merge this two lines, i.e. `for (volume, loanPeriod, itemList, noOfItems, recordNo) in result`

File: modules/bibcirculation/lib/bibcirculationadminlib.py [hunk 7]
```diff
@@ -3075,6 +3075,82 @@ def update_item_info_step6(req, tup_infos, ln=CFG_SITE_LANG):
...
+    body = bc_templates.tmpl_loan_period_calculate(infos=infos, ln=ln)
+
+    return page(title="Loan Period Calculate",
+                uid=id_user,
+                req=req,
+                body=body,
+                navtrail=navtrail_previous_links,
+                lastupdated=__lastupdated__)
+
+def loan_period_result(req, s, e, n, ln=CFG_SITE_LANG):
```
Comment (egabancho) [inline modules/bibcirculation/lib/bibcirculationadminlib.py:3104]: Same as https://github.com/CERNDocumentServer/invenio/pull/132/files?diff=unified#r36193559

File: modules/bibcirculation/web/admin/bibcirculationadmin.py [hunk 8]
```diff
@@ -71,6 +71,21 @@ def item_search_result(req, p=None, f=None, ln=CFG_SITE_LANG):
...
     return bal.item_search_result(req, p, f, ln)
 
+
+def loan_period_calculate(req, ln=CFG_SITE_LANG):
+    """
+    http://cdsweb.cern.ch/admin/bibcirculation/bibcirculationadmin.py/loan_period_calculate
+    """
+    return bal.loan_period_calculate(req, ln)
+
+def loan_period_result(req, s=None, e=None, n=None, ln=CFG_SITE_LANG):
```
Comment (egabancho) [inline modules/bibcirculation/web/admin/bibcirculationadmin.py:81]: Same as https://github.com/CERNDocumentServer/invenio/pull/132/files?diff=unified#r36193559

**Comment by egabancho** [comment]
Can you make this pull request on top of invenio:legacy instead of our operations branch? And also make a pull request to invenio itself?

cc @mvesper

**Comment by chakshuahuja** [comment]
Done with the changes. So can you just send me the link of both repos to which the pull request is to be made? I ll delete this pull request to save confusion.

**Comment by egabancho** [comment]
You don't need to delete this PR, just do `git push origin <your-branch> --force` once you do the changes and the pull request should get updated actomagically.

Abut the repos, this in the link to Invenio legacy branch https://github.com/inveniosoftware/invenio/tree/legacy

**Comment by chakshuahuja** [comment]
Sent the pull request to invenio legacy branch after making the changes.
Once that is merged and there are  no more changes to be made, I ll send to
this as well to update it here.

**Comment by egabancho** [comment]
Can you make the PR at the same time in both places? We don't know when it is going to get integrated in Invenio upstream and I assume we want this asap on CDS.

You can ask @mvesper how to do this, he did it already in the past.

**Comment by chakshuahuja** [comment]
Sent separately to both the repos. Will that be fine?

**Comment by egabancho** [comment]
OK, no worries

**Comment by egabancho** [comment]
Superseded by: #133

**Pull request closed without merging**